.github/workflows/tests-contributors.yml

```diff
@@ -37,30 +37,77 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Generate matrix for test jobs
+        id: generate_matrix
+        run: |
+          FEDVERS=34
+          TARGET_BRANCH=${{ fromJson(steps.pr_api.outputs.data).base.ref }}
+
+          JSON_MASTER_ELN='{"release": "master", "target-branch": "master", "ci-tag": "master"}, {"release": "eln", "target-branch": "master", "ci-tag": "eln", "build-args": "--build-arg=image=quay.io/fedoraci/fedora:eln-x86_64"}'
+          JSON_DEVEL='{"release": "f'$FEDVERS'-devel", "target-branch": "f'$FEDVERS'-devel", "ci-tag": "f'$FEDVERS'-devel"}'
+          JSON_RELEASE='{"release": "f'$FEDVERS'-release", "target-branch": "f'$FEDVERS'-release", "ci-tag": "f'$FEDVERS'-release"}'
+
+          REF_DEVEL='f'$FEDVERS'-devel'
+          REF_RELEASE='f'$FEDVERS'-release'
+
+          # Always order things so that first array element is the current branch, because... wait for it...
+          if [[ "$TARGET_BRANCH" == "$REF_DEVEL" ]] ; then
+            MERGED='{"include": ['$JSON_DEVEL', '$JSON_RELEASE', '$JSON_MASTER_ELN']}' ;
+          elif [[ "$TARGET_BRANCH" == "$REF_RELEASE" ]] ; then
+            MERGED='{"include": ['$JSON_RELEASE']}' ;
+          elif [[ "$TARGET_BRANCH" == "master" ]] ; then
+            MERGED='{"include": ['$JSON_MASTER_ELN']}' ;
+          else
+            OUT_JSON_FULL=''
+            OUT_JSON_PRIMARY=''
+            OUT_JSON_SECONDARY=''
+          fi
+
+          if [[ -z $OUT_JSON_FULL ]] ; then
+            # ...here "release" in first array element is replaced with empty string to make the displayed name always the same.
+            OUT_JSON_FULL=$(echo "$MERGED" | jq -c '.include[0].release |= ""')
+            # Turn "release" values into an array to be used as the primary key for matrix. Do not use the whole JSON directly, it generates FUGLY job names.
+            OUT_JSON_PRIMARY=$(echo "$OUT_JSON_FULL" | jq -c '[.include[].release]')
```

Comment:
I don't think you want to complicate things by this. Add everything to include and avoid having the `release:` part completely.

Reply:
Unfortunately that does not work. So we have to choose - required checks with two matrix variables, or one matrix variable and no required checks. (I like the first option better)

Why: Name of the job is automatically created from the constant in yaml and variables in matrix. If we do just `include` then there is no "primary variable" and everything goes into the status name. There is also no way to set *job* name dynamically (job != step).

Reply:
Resolved in next round of review below.